### PR TITLE
Report the proper location for errors compiled in templates

### PR DESCRIPTION
**Pull request by stof**
The `{% use %}` and `{% with %}` tags are adding some runtime checks triggering exceptions in the compiled template. This ensures that they get the proper location.
While the guessing was generally working fine for the `{% with %}` (and so this only makes the code faster), the guessing was not working for `{% use %}` due to the exception happening in the class constructor rather than on display (and so the guessing was finding the template which was triggering the load of the faulty template).

### CODE REVIEW

**Comment by stof** [comment]
I will send a separate PR to 1.x, as it does not have `$this->source` in compiled templates AFAIK

File: lib/Twig/Node/Module.php [hunk 1]
```diff
@@ -210,11 +212,13 @@ protected function compileConstructor(Twig_Compiler $compiler)
                         ->string($key)
                         ->raw("])) {\n")
                         ->indent()
-                        ->write("throw new Twig_Error_Runtime(sprintf('Block ")
+                        ->write("throw new Twig_Error_Runtime('Block ")
                         ->string($key)
                         ->raw(' is not defined in trait ')
                         ->subcompile($trait->getNode('template'))
-                        ->raw(".'));\n")
+                        ->raw(".',")
```
Comment (fabpot) [inline lib/Twig/Node/Module.php]: missing space after the comma

File: lib/Twig/Node/With.php [hunk 2]
```diff
@@ -38,7 +38,9 @@ public function compile(Twig_Compiler $compiler)
                 ->raw(";\n")
                 ->write(sprintf("if (!is_array(\$%s)) {\n", $varsName))
                 ->indent()
-                ->write("throw new Twig_Error_Runtime('Variables passed to the \"with\" tag must be a hash.');\n")
+                ->write("throw new Twig_Error_Runtime('Variables passed to the \"with\" tag must be a hash.',")
```
Comment (fabpot) [inline lib/Twig/Node/With.php]: missing space after the comma

File: lib/Twig/Node/Module.php [hunk 3]
```diff
@@ -198,7 +198,9 @@ protected function compileConstructor(Twig_Compiler $compiler)
                     ->indent()
                     ->write("throw new Twig_Error_Runtime('Template \"'.")
                     ->subcompile($trait->getNode('template'))
-                    ->raw(".'\" cannot be used as a trait.');\n")
+                    ->raw(".'\" cannot be used as a trait.',")
```
Comment (fabpot) [inline lib/Twig/Node/Module.php]: missing space after the comma

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by stof** [comment]
@fabpot fixed.

**Comment by fabpot** [comment]
Thank you @stof.

**Pull request merged**